### PR TITLE
Create new schema JSON files

### LINKED ISSUE #11: Add support for authoring tool 1

### PR DESCRIPTION
**Pull request by zubairslamdien**
Fixes #11

### CODE REVIEW

File: schema/component.schema.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
+  "$anchor": "_hint-component",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "$patch": {
```
Comment (chris-steele) [inline schema/component.schema.json:5]: I think this should be $merge for component schemas?

Comment (oliverfoster) [inline schema/component.schema.json:5]: https://adapt-security.github.io/adapt-authoring-documentation/manual/#/writing-a-schema?id=defining-schema-inheritance

**Comment by zubairslamdien** [comment]
Thanks @oliverfoster I've changed that now.

**Comment by chris-steele** [comment]
@zubairslamdien I think you were correct with `$patch`, it was my mistake to suggest a change. Is that right @oliverfoster ?

**Comment by oliverfoster** [comment]
Yeas, I think so @chris-steele , @zubairslamdien was correct with `$patch`

`$patch` seems to be for extension schemas, `$merge` is for the hierarchy/content schemas. So you'd use `$merge` to do the inheritance/extension from some other named schema to make a new named schema. Whereas `$patch` is for modifying an existing named schema.

In this case it's a tad confusing because we're using `$patch` to modify the existing `component` schema which is inherited/extended by other component plugin schemas.

**Pull request merged**